sci-mathematics/cvc3/cvc3-2.4.1-r1.ebuild

```diff
@@ -0,0 +1,158 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI="6"
+
+inherit elisp-common
+
+DESCRIPTION="CVC3 is a theorem prover for Satisfiability Modulo Theories (SMT) problems"
+HOMEPAGE="http://www.cs.nyu.edu/acsys/cvc3/index.html"
+SRC_URI="http://www.cs.nyu.edu/acsys/cvc3/releases/2.4.1/${P}.tar.gz"
+
+LICENSE="BSD MIT HPND zchaff? ( zchaff )"
+RESTRICT="mirror zchaff? ( bindist )"
+SLOT="0/${PV}"
+KEYWORDS="~amd64 ~x86"
+IUSE="doc emacs isabelle test zchaff"
+
+RDEPEND="dev-libs/gmp:0=
+		isabelle? (
+			>=sci-mathematics/isabelle-2011.1-r1:=
+		)"
+DEPEND="${RDEPEND}
+		doc? (
+			app-doc/doxygen
+			media-gfx/graphviz
+		)
+		emacs? (
+			virtual/emacs
+		)"
+
+SITEFILE=50${PN}-gentoo.el
+
+PATCHES="${FILESDIR}/${P}-gccv6-fix.patch"
+
+src_prepare() {
+	sed -e 's#prefix=@prefix@#prefix=$(patsubst %/,%,$(DESTDIR))@prefix@#' \
+		-e 's#libdir=@libdir@#libdir=$(patsubst %/,%,$(DESTDIR))@libdir@#' \
+		-e 's#mandir=@mandir@#mandir=$(patsubst %/,%,$(DESTDIR))@mandir@#' \
+		-i "${S}/Makefile.local.in" \
+		|| die "Could not set DESTDIR in Makefile.local.in"
+
+	default
+}
+
+src_configure() {
+	# --enable-static disables building of shared libraries, statically
+	# links /usr/bin/cvc3 and installs static libraries.
+	# --enable-static --enable-sharedlibs behaves the same as just --enable-static
+	econf \
+		--enable-dynamic \
+		$(use_enable zchaff)
+
+	if use test; then
+		sed -e 's@LD_LIBS = @LD_LIBS = -L'"${S}"'/lib -Wl,-R'"${S}"'/lib @' \
+			-i "${S}/test/Makefile" \
+			|| die "Could not set library paths in test/Makefile"
+	fi
+}
+
+src_compile() {
+	emake
+
+	if use doc; then
+		pushd doc || die "Could not cd to doc"
+		emake
+		popd
+	fi
+
+	if use emacs ; then
+		pushd "${S}/emacs" || die "Could change directory to emacs"
+		elisp-compile *.el || die "emacs elisp compile failed"
+		popd
+	fi
+
+	if use test; then
+		pushd test || die "Could not cd to test"
+		emake
+		popd
+	fi
+}
+
+src_test() {
+	pushd test || die "Could not cd to test"
+	./bin/test || die "tests failed"
+	popd
+}
+
+src_install() {
+	emake DESTDIR="${D}" install
+
+	if use doc; then
+		HTML_DOCS="${S}/doc/html/*.html ${S}/doc/html/*.gif ${S}/doc/html/*.png ${S}/doc/html/*.css"
```

Comment:
This worked for me currently.  Switching to your recommendation works, but it pulls in several other files that I don't think are necessary (such as md5 hashes of the graphs).  If that isn't a problem, I'll switch it over.

All your other suggestions look good, and I'm working through them while trying to test everything.